src/main/kotlin/org/stellar/walletsdk/Exception.kt

```diff
@@ -0,0 +1,57 @@
+package org.stellar.walletsdk
+
+import java.io.IOException
+import java.math.BigDecimal
+import okhttp3.Response
+import org.stellar.sdk.responses.SubmitTransactionResponse
+
+class AccountNotEnoughBalanceException(
+  accountAddress: String,
+  accountBalance: BigDecimal,
+  transactionFees: BigDecimal,
+) :
+  Exception(
+    "Source account $accountAddress does not have enough XLM balance to cover " +
+      "${transactionFees.toPlainString()} XLM fees. Available balance ${accountBalance.toPlainString()} XLM."
+  )
+
+class AccountNotFoundException(accountAddress: String) :
+  Exception("Account $accountAddress was not found")
+
+class ClientDomainWithMemoException : Exception("Client domain cannot be used with memo")
+
+class InvalidMemoIdException : Exception("Memo ID must be a positive integer")
+
+class InvalidSponsorOperationTypeException(operationType: String, allowedOperations: List<String>) :
+  Exception(
+    "$operationType cannot be sponsored. Allowed operations are: ${allowedOperations
+      .joinToString(", ")}."
+  )
+
+class InvalidStartingBalanceException :
+  Exception("Starting balance must be at least 1 XLM for non-sponsored accounts")
```

Comment:
this might be too in the weeds for this use case, but is it worth considering subentries for minimum balance?

https://developers.stellar.org/docs/fundamentals-and-concepts/lumens#base-reserves

Reply:
This exception is for an invalid minimum balance during account creation. We have a method that calculates reserved balance, which takes into account subentries. Is there a use case where we would need to calculate subentries for account creation (setting starting account balance)? Or maybe the exception name is not clear for the intended use case? :thinking:

Reply:
Oh got it. I didn't quite understand the context of this being during `fund` during Account Creation, but looking at where this is thrown, it makes more sense now. 👍